wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Forward.swift

```diff
@@ -96,7 +96,7 @@ extension ZMMessage: Shareable {
             }
         } else if isVideo || isAudio || isFile {
             guard let url = fileMessageData!.temporaryURLToDecryptedFile() else { return }
-            FileMetaDataGenerator.metadataForFileAtURL(url, UTI: url.UTI(), name: url.lastPathComponent) { fileMetadata in
+            FileMetaDataGenerator.shared.metadataForFileAtURL(url, UTI: url.UTI(), name: url.lastPathComponent) { fileMetadata in
```

Comment:
There are only two places that use the deprecated singleton `FileMetaDataGenerator.shared`. Can we change it to instantiate an object right in place `FileMetaDataGenerator`, as you already did in some changes in this PR? 👍

And FileMetaDataGenerator can become a `struct`.

Reply:
If I instantiate the object in place, it risks being deallocated before being able to call the completion block. And since we're in an extension, I cannot save it as a property